packages/v4-sdk/src/utils/math/shiftDiv.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-bitwise */
+import invariant from 'tiny-invariant'
+
+export const shiftDivRoundDown = (x: bigint, offset: bigint | number, denominator: bigint): bigint => {
+  const prod0 = x << BigInt(offset)
+  const prod1 = x >> (256n - BigInt(offset))
+
+  return _getEndOfDivRoundDown(x, 1n << BigInt(offset), denominator, prod0, prod1)
```

Comment:
I'm thinking to simplify this part. As we don't need to deal with overflow in js with bigint.  Let's simply do the division and round up